ldap/src/test/java/org/springframework/security/ldap/authentication/ad/ActiveDirectoryLdapAuthenticationProviderTests.java

```diff
@@ -250,100 +239,72 @@ public void passwordNeedsResetIsCorrectlyMapped() {
 		this.provider.contextFactory = createContextFactoryThrowing(new AuthenticationException(msg + "533, xxxx]"));
 		this.provider.setConvertSubErrorCodesToExceptions(true);
-		this.provider.authenticate(this.joe);
+		assertThatExceptionOfType(DisabledException.class).isThrownBy(() -> this.provider.authenticate(this.joe));
 	}
 
-	@Test(expected = AccountExpiredException.class)
+	@Test
 	public void accountExpiredIsCorrectlyMapped() {
 		this.provider.contextFactory = createContextFactoryThrowing(new AuthenticationException(msg + "701, xxxx]"));
 		this.provider.setConvertSubErrorCodesToExceptions(true);
-		this.provider.authenticate(this.joe);
+		assertThatExceptionOfType(AccountExpiredException.class).isThrownBy(() -> this.provider.authenticate(this.joe));
 	}
 
-	@Test(expected = LockedException.class)
+	@Test
 	public void accountLockedIsCorrectlyMapped() {
 		this.provider.contextFactory = createContextFactoryThrowing(new AuthenticationException(msg + "775, xxxx]"));
 		this.provider.setConvertSubErrorCodesToExceptions(true);
-		this.provider.authenticate(this.joe);
+		assertThatExceptionOfType(LockedException.class).isThrownBy(() -> this.provider.authenticate(this.joe));
 	}
 
-	@Test(expected = BadCredentialsException.class)
+	@Test
 	public void unknownErrorCodeIsCorrectlyMapped() {
 		this.provider.contextFactory = createContextFactoryThrowing(new AuthenticationException(msg + "999, xxxx]"));
 		this.provider.setConvertSubErrorCodesToExceptions(true);
-		this.provider.authenticate(this.joe);
+		assertThatExceptionOfType(BadCredentialsException.class).isThrownBy(() -> this.provider.authenticate(this.joe));
 	}
 
-	@Test(expected = BadCredentialsException.class)
+	@Test
 	public void errorWithNoSubcodeIsHandledCleanly() {
 		this.provider.contextFactory = createContextFactoryThrowing(new AuthenticationException(msg));
 		this.provider.setConvertSubErrorCodesToExceptions(true);
-		this.provider.authenticate(this.joe);
+		assertThatExceptionOfType(BadCredentialsException.class).isThrownBy(() -> this.provider.authenticate(this.joe));
 	}
 
-	@Test(expected = org.springframework.ldap.CommunicationException.class)
+	@Test
 	public void nonAuthenticationExceptionIsConvertedToSpringLdapException() throws Throwable {
-		try {
+		assertThatExceptionOfType(InternalAuthenticationServiceException.class).isThrownBy(() -> {
 			this.provider.contextFactory = createContextFactoryThrowing(new CommunicationException(msg));
 			this.provider.authenticate(this.joe);
-		}
-		catch (InternalAuthenticationServiceException ex) {
-			// Since GH-8418 ldap communication exception is wrapped into
-			// InternalAuthenticationServiceException.
-			// This test is about the wrapped exception, so we throw it.
-			throw ex.getCause();
-		}
+		}).withCauseInstanceOf(org.springframework.ldap.CommunicationException.class);
 	}
 
-	@Test(expected = org.springframework.security.authentication.InternalAuthenticationServiceException.class)
```

Comment:
Should this instead be replaced with `@Test`?

Reply:
Yes, I must have accidentally deleted it. I'll push a fix then we can rebased when all the other issues are addressed.